src/H5private.h

```diff
@@ -1111,14 +1093,37 @@ extern char H5_lib_vers_info_g[];
 
 #endif /* H5_HAVE_THREADSAFE */
 
-/* Library init / term status (global) */
-extern bool H5_libinit_g; /* Has the library been initialized? */
-extern bool H5_libterm_g; /* Is the library being shutdown? */
-
 /* Macros for accessing the global variables */
 #define H5_INIT_GLOBAL (H5_libinit_g)
 #define H5_TERM_GLOBAL (H5_libterm_g)
 
+/* Macros for referencing package initialization symbols */
+#define H5_PACKAGE_INIT_VAR(x)  H5_GLUE(x, _init_g)
+#define H5_PACKAGE_INIT_FUNC(x) H5_GLUE(x, __init_package)
+
+/* Macros for defining package initialization routines */
+#ifdef H5_MY_PKG
+#define H5_PKG_INIT_VAR  H5_PACKAGE_INIT_VAR(H5_MY_PKG)
+#define H5_PKG_INIT_FUNC H5_PACKAGE_INIT_FUNC(H5_MY_PKG)
+#define H5_PACKAGE_YES_INIT(err)                                                                             \
+    /* Initialize this interface or bust */                                                                  \
+    if (H5_UNLIKELY(!H5_PKG_INIT_VAR && !H5_TERM_GLOBAL)) {                                                  \
+        H5_PKG_INIT_VAR = true;                                                                              \
+        if (H5_PKG_INIT_FUNC() < 0) {                                                                        \
+            H5_PKG_INIT_VAR = false;                                                                         \
+            HGOTO_ERROR(H5E_FUNC, H5E_CANTINIT, err, "interface initialization failed");                     \
+        }                                                                                                    \
+    }
+#define H5_PACKAGE_NO_INIT(err)                                                                              \
+    /* Initialize this interface or bust */                                                                  \
```

Comment:
It would be more accurate to say "flag this interface as trivially initialized", since this can't throw an error and is for modules without actual initialization work (if I understand `H5_MY_PKG_INIT` correctly.)

Reply:
I'll update the comment.